hooks/wallet/use-axone-payments.tsx

```diff
@@ -150,7 +151,233 @@ export const useAxonePayments = () => {
 
     setBalanceDenom(balance.balance?.denom || 'know');
     setBalance(amount, false);
-  }, [setBalance, setBalanceDenom]);
+  }, [getRpcEndpoint, isDev, setBalance, setBalanceDenom]);
+
+  const delegateToValidator = async ({ amount, validatorAddress, memo }: { amount: number, validatorAddress: string, memo: string}) => {
+    setIsTransactionPending(true);
+    const stargateClient = await getSigningStargateClient();
+    if (!stargateClient || !address) {
+      console.error('stargateClient undefined or address undefined.');
+      return;
+    }
+
+    const { delegate } = cosmos.staking.v1beta1.MessageComposer.withTypeUrl;
+
+    const msg = delegate({
+      delegatorAddress: address,
+      validatorAddress: validatorAddress,
+      amount: {
+        denom: coin.base,
+        amount: new BigNumber(amount).multipliedBy(1 ** 6).toString(),
+      }
+    });
+
+    const fee: StdFee = {
+      amount: [
+        {
+          denom: coin.base,
+          amount: '1000',
+        },
+      ],
+      gas: '200000',
+    };
+
+    try {
+      await stargateClient.signAndBroadcast(
+        address,
+        [msg],
+        fee,
+        memo
+      );
+      toast({
+        action: (
+          <Row className='items-center -ml-2'>
```

Comment:
Looks like you are reusing this layout a lot of times.
Consider to create a component for it